Source/ARTLocalDevice.m

```diff
@@ -72,24 +71,18 @@ + (ARTLocalDevice *)load:(NSString *)clientId storage:(id<ARTDeviceStorage>)stor
     #endif
     device.push.recipient[@"transportType"] = ARTDevicePushTransportType;
 
-    NSString *deviceId = [storage objectForKey:ARTDeviceIdKey];
+    NSString *deviceId = [storage objectForKey:ARTDeviceIdKey] ?: @""; // PCD2, not nullable
```

Comment:
I guess statement below that checks for `nil` will be affected. I think we shouldn't replace `nil` with empty string just because in the spec id is not nullable.

Reply:
Does it mean it should become nullable in the spec? Or should we ignore the spec because an empty string check in objective-c looks ugly?

Reply:
I think it should be nullable in spec. It's not because it's ugly, it's because it's inconsistent. If id is not set we use `nil` everywhere, what is the reason to do differently here?

Reply:
>I think it should be nullable in spec.
Agree. Will change it back then. Nice catch!

Reply:
Thank you!

Reply:
It will be a breaking change tho... I guess we should discuss it with the rest of the team and this shouldn't be done in this PR.

Reply:
Lets rely on what already was approved by Lawrence https://github.com/ably/ably-cocoa/pull/1259

Reply:
https://github.com/ably/ably-cocoa/pull/1847/commits/8381e102e261e302cb3e986c53f3971ecd9b2e5b